metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -0,0 +1,114 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import coursierapi.error.SimpleResolutionError
+
+trait MtagsResolver {
+  def resolve(scalaVersion: String): Option[MtagsBinaries]
+  final def isSupportedScalaVersion(version: String): Boolean =
+    resolve(version).isDefined
+}
+
+object MtagsResolver {
+
+  def default(): MtagsResolver = new Default
+
+  /**
+   * The default MtagsResolver always checks if mtags-artifacts for particular scala version exists.
+   * However this strict logic doesn't work for tests.
+   * For `unit` we don't publish mtags at all but there are some tests that trigger docker check.
+   * So for these cases, do fallback on previous mechanic by checking declared supported versions.
+   */
+  def forUnitTests(): MtagsResolver =
```

Comment:
Anyway I had to add this for unit tests.
There is a suite that check if messages for support/unsupported versions are produced correctly and more other.

Moving it in slow won't solve the problem as we don't publish only specfic range of versions for these tests.